server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -1924,6 +1931,60 @@ private Page<Candidate> doSearchCandidates(SearchCandidateRequest searchRequest)
 
         return candidates;
     }
+
+    /**
+     * Search explicitly by candidateNumber.
+     *
+     * - Accepts a list of candidateNumber strings from the request.
+     * - Fetches all matching candidates using repository method:
+     *      findByCandidateNumberIn(Collection<String>, Pageable)
+     * - Removes excluded candidates.
+     * - Preserves the exact order in which numbers were entered.
+     */
+    private Page<Candidate> searchByCandidateNumbers(
+        SearchCandidateRequest request,
+        Set<Candidate> excludedCandidates
+    ) {
+        List<String> candidateNumbers = request.getCandidateNumbers();
+
+        // 1) No numbers → Caller fall back to normal search
+        if (candidateNumbers == null || candidateNumbers.isEmpty()) {
+            return null;
+        }
+
+        Page<Candidate> page = candidateRepository.findByCandidateNumberIn(
+            candidateNumbers,
+            PageRequest.of(request.getPageNumber(), request.getPageSize())
+        );
+
+        // 2) Remove excluded candidates
+        List<Candidate> filtered = page.getContent().stream()
+            .filter(c -> !excludedCandidates.contains(c))
+            .collect(Collectors.toList());
+
+        // 3) Maintain input order (important for user experience)
+        Map<String, Integer> orderByNumber = new HashMap<>();
```

Comment:
Search sorts are controlled by clicking on the results column. 

You can't assume that the user wants the results sorted by candidate number.